processing/src/main/java/io/druid/query/filter/LikeDimFilter.java

```diff
@@ -0,0 +1,358 @@
+            prefix.append(c);
+          } else {
+            // Excessively paranoid escaping, although shouldn't affect runtime beyond compiling the regex.
+            if (DEFINITELY_FINE.matcher(String.valueOf(c)).matches()) {
+              regex.append(c);
+            } else {
+              regex.append("\\u").append(BaseEncoding.base16().encode(Chars.toByteArray(c)));
+            }
+          }
+          escaping = false;
+        }
+      }
+
+      final String regexString = regex.toString();
+
+      if (regexString.isEmpty()) {
+        return new LikeMatcher(prefix.toString(), SuffixStyle.MATCH_EMPTY, null);
+      } else if (regexString.equals(".*")) {
+        return new LikeMatcher(prefix.toString(), SuffixStyle.MATCH_ANY, null);
+      } else {
+        return new LikeMatcher(prefix.toString(), SuffixStyle.MATCH_PATTERN, Pattern.compile(regexString));
+      }
+    }
+
+    public boolean matches(@Nullable final String s)
+    {
+      if (s == null) {
+        return prefix == null && (suffixStyle == SuffixStyle.MATCH_ANY
+                                  || suffixStyle == SuffixStyle.MATCH_EMPTY
+                                  || suffixPattern.matcher("").matches());
+      }
+
+      final String suffix;
+
+      if (prefix == null) {
+        suffix = s;
+      } else if (s.startsWith(prefix)) {
+        suffix = s.substring(prefix.length());
+      } else {
+        return false;
+      }
+
+      if (suffixStyle == SuffixStyle.MATCH_ANY) {
+        return true;
+      } else if (suffixStyle == SuffixStyle.MATCH_EMPTY) {
+        return suffix.isEmpty();
+      } else {
+        // suffixStyle is MATCH_PATTERN
+        return suffixPattern.matcher(suffix).matches();
+      }
+    }
+
+    public DruidPredicateFactory predicateFactory(final ExtractionFn extractionFn)
+    {
+      return new DruidPredicateFactory()
+      {
+        @Override
+        public Predicate<String> makeStringPredicate()
+        {
+          if (extractionFn != null) {
+            return new Predicate<String>()
+            {
+              @Override
+              public boolean apply(String input)
+              {
+                return matches(extractionFn.apply(input));
+              }
+            };
+          } else {
+            return new Predicate<String>()
+            {
+              @Override
+              public boolean apply(String input)
+              {
+                return matches(input);
+              }
+            };
+          }
+        }
+
+        @Override
+        public DruidLongPredicate makeLongPredicate()
+        {
+          if (extractionFn != null) {
+            return new DruidLongPredicate()
+            {
+              @Override
+              public boolean applyLong(long input)
+              {
+                return matches(extractionFn.apply(input));
+              }
+            };
+          } else {
+            return new DruidLongPredicate()
```

Comment:
Ok anyway it should (?) be changed in a number of `DimFilter`s, so when and if I will make such PR this could be discussed again